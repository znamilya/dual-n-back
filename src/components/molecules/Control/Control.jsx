import React, { PropTypes } from 'react';
import bemCl from 'bem-cl';

import HotKey from 'components/atoms/HotKey/HotKey';
import AnswerResult from 'components/atoms/AnswerResult/AnswerResult';


const b = bemCl('control');

class Control extends React.Component {

    /* ------------------------------------------------------------------------------------------ */
    /* RENDER                                                                                     */
    /* ------------------------------------------------------------------------------------------ */
    render() {
        return (
            <div className={b()}>
                <HotKey {...this.props} />
                <AnswerResult {...this.props} />
            </div>
        );
    }
}


export default Control;
