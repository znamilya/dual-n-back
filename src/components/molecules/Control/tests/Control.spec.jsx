import React from 'react';
import { shallow } from 'enzyme';

import Control from '../Control';


jest.mock('components/atoms/HotKey/HotKey', () => 'HotKey');
jest.mock('components/atoms/AnswerResult/AnswerResult', () => 'AnswerResult');


describe('<Control />', () => {
    it('default render', () => {
        const component = shallow(
            <Control />
        );

        expect(component).toMatchSnapshot();
    });
});
