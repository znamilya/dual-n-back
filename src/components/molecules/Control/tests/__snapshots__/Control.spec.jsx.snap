exports[`<Control /> default render 1`] = `
<div
  className="control">
  <HotKey />
  <AnswerResult />
</div>
`;
